File: DAGs/libs/algorithm/poh_metrics/poh_metrics/tests/test_metrics.py
# D:\city_chain_project\DAGs\libs\algorithm\poh_metrics\poh_metrics\tests\test_metrics.py

import pytest
from prometheus_client import CollectorRegistry

from poh_metrics.metrics import register_metrics


def test_metrics_registered():
    reg = CollectorRegistry()
    register_metrics(reg)

    # .collect() で返る MetricFamily 名は _total のつかない名前！
    names = {mf.name for mf in reg.collect()}

    expected = {
        "poh_issued",
        "poh_issue_latency_seconds",
        "poh_issue_latency_summary_seconds",
        "poh_verify_latency_seconds",
        "poh_verify_latency_summary_seconds",
        "gc_events",
        "active_peers",
        "http_requests",
        "grpc_requests",
    }

    assert expected.issubset(names)

File: DAGs/libs/algorithm/poh_holdmetrics/poh_holdmetrics_rust/tests/test_metrics.rs
// \city_chain_project\DAGs\libs\algorithm\poh_holdmetrics\poh_holdmetrics_rust\tests\test_metrics.rs
//! Prometheus gather テスト：メトリクスが 1 つ以上登録済みか
use poh_holdmetrics_rust::metrics;

#[test]
fn prometheus_metrics_exist() {
    // アクセスすると counters が初期化されて gather に出てくる
    metrics::HOLD_EVENTS.inc();
    let gathered = prometheus::gather();
    let names: Vec<_> = gathered.iter().map(|m| m.get_name()).collect();
    assert!(names.contains(&"hold_events_total"));
}
